// media-chrome/dist/experimental/media-chrome-selectmenu.js
var __accessCheck = (obj, member, msg) => {
  if (!member.has(obj))
    throw TypeError("Cannot " + msg);
};
var __privateGet = (obj, member, getter) => {
  __accessCheck(obj, member, "read from private field");
  return getter ? getter.call(obj) : member.get(obj);
};
var __privateAdd = (obj, member, value) => {
  if (member.has(obj))
    throw TypeError("Cannot add the same private member more than once");
  member instanceof WeakSet ? member.add(obj) : member.set(obj, value);
};
var __privateSet = (obj, member, value, setter) => {
  __accessCheck(obj, member, "write to private field");
  setter ? setter.call(obj, value) : member.set(obj, value);
  return value;
};
var __privateMethod = (obj, member, method) => {
  __accessCheck(obj, member, "access private method");
  return method;
};
var _mediaController, _handleButtonClick, _handleOptionChange, _enabledState, _button, _buttonSlot, _listbox, _listboxSlot, _expanded, _keyupListener, _keydownListener, _documentClickHandler, _handleButtonClick_, handleButtonClick__fn, _handleOptionChange_, handleOptionChange__fn, _toggle, toggle_fn, _updateMenuPosition, updateMenuPosition_fn, _toggleExpanded, toggleExpanded_fn;
import "../media-chrome-button.js";
import "./media-chrome-listbox.js";
import { globalThis, document } from "../utils/server-safe-globals.js";
import { closestComposedNode, getOrInsertCSSRule } from "../utils/element-utils.js";
import { MediaStateReceiverAttributes } from "../constants.js";
const template = document.createElement("template");
template.innerHTML = `
  <style>
  :host {
    font: var(--media-font,
      var(--media-font-weight, normal)
      var(--media-font-size, 15px) /
      var(--media-text-content-height, var(--media-control-height, 24px))
      var(--media-font-family, helvetica neue, segoe ui, roboto, arial, sans-serif));
    color: var(--media-text-color, var(--media-primary-color, rgb(238 238 238)));
    display: inline-flex;
    position: relative;
    flex-shrink: .5;
  }

  [name="listbox"]::slotted(*),
  [part=listbox] {
    position: absolute;
    left: 0;
    bottom: 100%;
    max-height: 300px;
    overflow: hidden auto;
    transition: var(--media-selectmenu-transition-in,
      visibility 0s, transform .15s ease-out, opacity .15s ease-out);
    transform: var(--media-listbox-transform-in, translateY(0) scale(1));
    visibility: visible;
    opacity: 1;
  }

  [name="listbox"][hidden]::slotted(*),
  [hidden] [part=listbox] {
    transition: var(--media-selectmenu-transition-out,
      visibility .15s ease-out, transform .15s ease-out, opacity .15s ease-out);
    transform: var(--media-listbox-transform-out, translateY(2px) scale(.99));
    visibility: hidden;
    opacity: 0;
    pointer-events: none;
  }

  slot[name="listbox"][hidden] {
    display: block;
  }
  </style>

  <slot name="button">
    <media-chrome-button aria-haspopup="listbox" part="button">
      <slot name="button-content"></slot>
    </media-chrome-button>
  </slot>
  <slot name="listbox" hidden>
    <media-chrome-listbox id="listbox" part="listbox">
      <slot></slot>
    </media-chrome-listbox>
  </slot>
`;
class MediaChromeSelectMenu extends globalThis.HTMLElement {
  constructor() {
    var _a;
    super();
    __privateAdd(this, _handleButtonClick_);
    __privateAdd(this, _handleOptionChange_);
    __privateAdd(this, _toggle);
    __privateAdd(this, _updateMenuPosition);
    __privateAdd(this, _toggleExpanded);
    __privateAdd(this, _mediaController, void 0);
    __privateAdd(this, _handleButtonClick, void 0);
    __privateAdd(this, _handleOptionChange, void 0);
    __privateAdd(this, _enabledState, true);
    __privateAdd(this, _button, void 0);
    __privateAdd(this, _buttonSlot, void 0);
    __privateAdd(this, _listbox, void 0);
    __privateAdd(this, _listboxSlot, void 0);
    __privateAdd(this, _expanded, false);
    __privateAdd(this, _keyupListener, (e) => {
      const { key } = e;
      if (!this.keysUsed.includes(key)) {
        this.removeEventListener("keyup", __privateGet(this, _keyupListener));
        return;
      }
      const isButton = e.composedPath().includes(__privateGet(this, _button));
      if (isButton && (key === "Enter" || key === " ")) {
        __privateGet(this, _handleButtonClick).call(this);
      } else if (key === "Escape" && !__privateGet(this, _listboxSlot).hidden) {
        __privateMethod(this, _toggle, toggle_fn).call(this);
      }
    });
    __privateAdd(this, _keydownListener, (e) => {
      const { metaKey, altKey, key } = e;
      if (metaKey || altKey || !this.keysUsed.includes(key)) {
        this.removeEventListener("keyup", __privateGet(this, _keyupListener));
        return;
      }
      e.preventDefault();
      this.addEventListener("keyup", __privateGet(this, _keyupListener), { once: true });
    });
    __privateAdd(this, _documentClickHandler, (e) => {
      if (e.composedPath().includes(this))
        return;
      if (!__privateGet(this, _listboxSlot).hidden) {
        __privateMethod(this, _toggle, toggle_fn).call(this);
      }
    });
    if (!this.shadowRoot) {
      this.attachShadow({ mode: "open" });
      this.shadowRoot.appendChild(template.content.cloneNode(true));
    }
    const { style } = getOrInsertCSSRule(this.shadowRoot, ":host");
    style.setProperty("display", `var(--media-control-display, var(--${this.localName}-display, inline-flex))`);
    __privateSet(this, _handleButtonClick, __privateMethod(this, _handleButtonClick_, handleButtonClick__fn).bind(this));
    __privateSet(this, _handleOptionChange, __privateMethod(this, _handleOptionChange_, handleOptionChange__fn).bind(this));
    (_a = this.init) == null ? void 0 : _a.call(this);
    __privateSet(this, _button, this.shadowRoot.querySelector("[part=button]"));
    __privateSet(this, _listbox, this.shadowRoot.querySelector("[part=listbox]"));
    __privateSet(this, _buttonSlot, this.shadowRoot.querySelector("slot[name=button]"));
    __privateGet(this, _buttonSlot).addEventListener("slotchange", () => {
      const newButton = __privateGet(this, _buttonSlot).assignedElements()[0];
      if (!newButton)
        return;
      this.disable();
      __privateSet(this, _button, newButton);
      __privateGet(this, _button).preventClick = true;
      if (__privateGet(this, _button).hasAttribute("disabled")) {
        __privateSet(this, _enabledState, false);
      }
      if (__privateGet(this, _enabledState)) {
        this.enable();
        __privateGet(this, _button).setAttribute("aria-haspopup", "listbox");
      } else {
        this.disable();
      }
    });
    __privateSet(this, _listboxSlot, this.shadowRoot.querySelector("slot[name=listbox]"));
    __privateGet(this, _listboxSlot).addEventListener("slotchange", () => {
      this.disable();
      __privateSet(this, _listbox, __privateGet(this, _listboxSlot).assignedElements()[0] || __privateGet(this, _listbox));
      this.enable();
    });
  }
  static get observedAttributes() {
    return [
      "disabled",
      MediaStateReceiverAttributes.MEDIA_CONTROLLER
    ];
  }
  enable() {
    __privateGet(this, _button).removeAttribute("disabled");
    __privateGet(this, _button).addEventListener("click", __privateGet(this, _handleButtonClick));
    __privateGet(this, _button).addEventListener("keydown", __privateGet(this, _keydownListener));
    __privateGet(this, _listbox).addEventListener("keydown", __privateGet(this, _keydownListener));
    __privateMethod(this, _toggleExpanded, toggleExpanded_fn).call(this);
    __privateGet(this, _listbox).addEventListener("change", __privateGet(this, _handleOptionChange));
    document.addEventListener("click", __privateGet(this, _documentClickHandler));
  }
  disable() {
    __privateGet(this, _button).setAttribute("disabled", "");
    __privateGet(this, _button).removeEventListener("click", __privateGet(this, _handleButtonClick));
    __privateGet(this, _button).removeEventListener("keydown", __privateGet(this, _keydownListener));
    __privateGet(this, _button).removeEventListener("keyup", __privateGet(this, _keyupListener));
    __privateGet(this, _listbox).removeEventListener("keydown", __privateGet(this, _keydownListener));
    __privateGet(this, _listbox).removeEventListener("keyup", __privateGet(this, _keyupListener));
    __privateGet(this, _listbox).addEventListener("change", __privateGet(this, _handleOptionChange));
    document.removeEventListener("click", __privateGet(this, _documentClickHandler));
  }
  attributeChangedCallback(attrName, oldValue, newValue) {
    var _a, _b, _c, _d, _e;
    if (attrName === MediaStateReceiverAttributes.MEDIA_CONTROLLER) {
      if (oldValue) {
        (_b = (_a = __privateGet(this, _mediaController)) == null ? void 0 : _a.unassociateElement) == null ? void 0 : _b.call(_a, this);
        __privateSet(this, _mediaController, null);
        __privateGet(this, _listbox).removeAttribute(MediaStateReceiverAttributes.MEDIA_CONTROLLER);
      }
      if (newValue) {
        __privateSet(this, _mediaController, (_c = this.getRootNode()) == null ? void 0 : _c.getElementById(newValue));
        (_e = (_d = __privateGet(this, _mediaController)) == null ? void 0 : _d.associateElement) == null ? void 0 : _e.call(_d, this);
        __privateGet(this, _listbox).setAttribute(MediaStateReceiverAttributes.MEDIA_CONTROLLER, newValue);
      }
    } else if (attrName === "disabled" && newValue !== oldValue) {
      if (newValue == null) {
        __privateSet(this, _enabledState, true);
        this.enable();
      } else {
        __privateSet(this, _enabledState, false);
        this.disable();
      }
    }
  }
  connectedCallback() {
    var _a, _b, _c;
    if (!this.hasAttribute("disabled")) {
      this.enable();
    }
    const mediaControllerId = this.getAttribute(
      MediaStateReceiverAttributes.MEDIA_CONTROLLER
    );
    if (mediaControllerId) {
      __privateSet(this, _mediaController, (_a = this.getRootNode()) == null ? void 0 : _a.getElementById(mediaControllerId));
      (_c = (_b = __privateGet(this, _mediaController)) == null ? void 0 : _b.associateElement) == null ? void 0 : _c.call(_b, this);
      __privateGet(this, _listbox).setAttribute(MediaStateReceiverAttributes.MEDIA_CONTROLLER, mediaControllerId);
    }
  }
  disconnectedCallback() {
    var _a, _b;
    this.disable();
    (_b = (_a = __privateGet(this, _mediaController)) == null ? void 0 : _a.unassociateElement) == null ? void 0 : _b.call(_a, this);
    __privateSet(this, _mediaController, null);
    __privateGet(this, _listbox).removeAttribute(MediaStateReceiverAttributes.MEDIA_CONTROLLER);
  }
  get keysUsed() {
    return ["Enter", "Escape", " ", "ArrowUp", "ArrowDown", "f", "c", "k", "m"];
  }
}
_mediaController = new WeakMap();
_handleButtonClick = new WeakMap();
_handleOptionChange = new WeakMap();
_enabledState = new WeakMap();
_button = new WeakMap();
_buttonSlot = new WeakMap();
_listbox = new WeakMap();
_listboxSlot = new WeakMap();
_expanded = new WeakMap();
_keyupListener = new WeakMap();
_keydownListener = new WeakMap();
_documentClickHandler = new WeakMap();
_handleButtonClick_ = new WeakSet();
handleButtonClick__fn = function() {
  __privateMethod(this, _toggle, toggle_fn).call(this);
};
_handleOptionChange_ = new WeakSet();
handleOptionChange__fn = function() {
  __privateMethod(this, _toggle, toggle_fn).call(this, true);
};
_toggle = new WeakSet();
toggle_fn = function(closeOnly) {
  __privateGet(this, _listboxSlot).hidden = !__privateGet(this, _listboxSlot).hidden || closeOnly;
  __privateMethod(this, _toggleExpanded, toggleExpanded_fn).call(this, closeOnly);
  if (!__privateGet(this, _listboxSlot).hidden) {
    __privateGet(this, _listbox).focus();
    __privateMethod(this, _updateMenuPosition, updateMenuPosition_fn).call(this);
  } else if (this.shadowRoot.activeElement === __privateGet(this, _listbox) || __privateGet(this, _listbox).contains(this.shadowRoot.activeElement)) {
    __privateGet(this, _button).focus();
  }
};
_updateMenuPosition = new WeakSet();
updateMenuPosition_fn = function() {
  var _a;
  if (__privateGet(this, _listbox).offsetWidth === 0)
    return;
  const buttonRect = __privateGet(this, _button).getBoundingClientRect();
  if (this.hasAttribute("mediacontroller") || __privateGet(this, _button).hasAttribute("mediacontroller") || __privateGet(this, _listbox).hasAttribute("mediacontroller")) {
    __privateGet(this, _listbox).style.zIndex = "1";
    __privateGet(this, _listbox).style.bottom = "unset";
    __privateGet(this, _listbox).style.top = buttonRect.height + "px";
    return;
  }
  const bounds = (_a = this.getAttribute("bounds") ? closestComposedNode(this, `#${this.getAttribute("bounds")}`) : this.parentElement) != null ? _a : this;
  const listboxWidth = __privateGet(this, _listbox).offsetWidth;
  const boundsRect = bounds.getBoundingClientRect();
  const position = -Math.max(buttonRect.x + listboxWidth - boundsRect.right, 0);
  __privateGet(this, _listbox).style.left = `${position}px`;
};
_toggleExpanded = new WeakSet();
toggleExpanded_fn = function(closeOnly = false) {
  __privateSet(this, _expanded, !__privateGet(this, _expanded) || closeOnly);
  __privateGet(this, _button).setAttribute("aria-expanded", __privateGet(this, _expanded));
};
if (!globalThis.customElements.get("media-chrome-selectmenu")) {
  globalThis.customElements.define("media-chrome-selectmenu", MediaChromeSelectMenu);
}
var media_chrome_selectmenu_default = MediaChromeSelectMenu;
export {
  MediaChromeSelectMenu,
  media_chrome_selectmenu_default as default
};
